udata/core/site/models.py

```diff
@@ -142,10 +142,15 @@ def count_max_org_datasets(self):
 def get_current_site():
     if getattr(g, 'site', None) is None:
         site_id = current_app.config['SITE_ID']
-        g.site, _ = Site.objects.get_or_create(id=site_id, defaults={
-            'title': current_app.config.get('SITE_TITLE'),
-            'keywords': current_app.config.get('SITE_KEYWORDS', []),
+        site_title = current_app.config.get('SITE_TITLE')
+        site_keywords = current_app.config.get('SITE_KEYWORDS', [])
+        g.site, created = Site.objects.get_or_create(id=site_id, defaults={
+            'title': site_title,
+            'keywords': site_keywords,
         })
+        if not created:
+            Site.objects(id=site_id).modify(set__title=site_title, set__keywords=site_keywords)
```

Comment:
This would imply a modification every time with access `current_site`, which may be overkill.
Instead of using a `not created` as condition which will be true most of the time, would it be more efficient to compare `if g.site.keyword != site_keyword` and same with title?

Reply:
True, I somehow thought that it won't be called every time.